x-pack/libbeat/reader/parquet/parquet_benchmark_test.go

```diff
@@ -0,0 +1,329 @@
+				rows: 100000,
+			}},
+			batchSize: 1000,
+			invokeFn:  readParquetFile,
+		},
+		{
+			desc:                       "Construct and read a single large parquet file in batches of 10000 serially",
+			constructAndReadLargeFiles: true,
+			largeFiles: []parquetFile{{
+				name: "largeFile_1.parquet",
+				cols: 4,
+				rows: 100000,
+			}},
+			batchSize: 10000,
+			invokeFn:  readParquetFile,
+		},
+	}
+
+	// cleanup process in case of abrupt exit
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+	go func() {
+		<-sigc
+		for _, tc := range testCases {
+			if tc.constructAndReadLargeFiles {
+				for _, f := range tc.largeFiles {
+					os.Remove(testDataPath + f.name)
+				}
+			}
+		}
+		os.Exit(1)
+	}()
+
+	for _, tc := range testCases {
+		b.Run(tc.desc, func(b *testing.B) {
+			cfg := &Config{
+				// we set ProcessParallel to true as this always has the best performance
+				ProcessParallel: true,
+				BatchSize:       tc.batchSize,
+			}
+
+			var checkMem bool
+			if tc.constructAndReadLargeFiles {
+				checkMem = true
+				for _, f := range tc.largeFiles {
+					fName := testDataPath + f.name
+					createRandomParquet(b, fName, f.cols, f.rows)
+					tc.files = append(tc.files, fName)
+					defer os.Remove(fName)
+				}
+			} else {
+				for i, f := range tc.files {
+					tc.files[i] = testDataPath + f
+				}
+			}
+
+			b.ResetTimer()
+			// in case of large files we do not want to report allocations here
+			// as it will skew the results. We report allocations in the invokeFn.
+			if !tc.constructAndReadLargeFiles {
+				b.ReportAllocs()
+			}
+			//nolint:errcheck // we do not care about handling errors from file.Seek()
+			switch {
+			case tc.processParallel:
+				b.RunParallel(func(pb *testing.PB) {
+					filePtrArr := openFiles(b, tc.files)
+					for pb.Next() {
+						for _, f := range filePtrArr {
+							defer f.Close()
+							f.Seek(0, 0)
+							tc.invokeFn(b, cfg, f, checkMem)
+						}
+					}
+				})
+			case tc.useGoRoutiunes:
+				filePtrArr := openFiles(b, tc.files)
+				wg := sync.WaitGroup{}
+				for i := 0; i < b.N; i++ {
+					for _, f := range filePtrArr {
+						defer f.Close()
+						f.Seek(0, 0)
+						cf := f
+						wg.Add(1)
+						go func() {
+							defer wg.Done()
+							tc.invokeFn(b, cfg, cf, checkMem)
+						}()
+					}
+					wg.Wait()
+				}
+			// default case is set to serial processing of files
+			default:
+				filePtrArr := openFiles(b, tc.files)
+				for i := 0; i < b.N; i++ {
+					for _, f := range filePtrArr {
+						defer f.Close()
+						f.Seek(0, 0)
+						tc.invokeFn(b, cfg, f, checkMem)
+					}
+				}
+			}
+		})
+	}
+}
+
+// readParquetFile reads entire parquet file
+func readParquetFile(b *testing.B, cfg *Config, file *os.File, checkMem bool) {
+	if checkMem {
+		b.ReportAllocs()
+	}
+	sReader, err := NewStreamReader(file, cfg)
+	if err != nil {
+		b.Fatalf("failed to init stream reader: %v\n", err)
```

Comment:
`\n` at the end in the format specifier of methods defined for *testing.B and *testing.T (that internally uses Logf) can be avoided because: 
> A final newline is added if not provided.

Ref: https://pkg.go.dev/testing#F.Logf